src/main/java/backend/Logic.java

```diff
@@ -271,6 +271,28 @@ public Model getRepo(String repoId) {
     }
 
     /**
+     * Replaces existing milestone with the newMilestone in the issue object, the UI, and the server, in that order.
+     * Server update is done after the local update to reduce the lag between the user action and the UI response
+     *
+     * @param issue The issue object whose milestone is to be replaced.
+     * @param newMilestone The new milestone to be assigned to the issue
+     * @return true if milestone replacement on GitHub was a success, false otherwise.
+     */
+    public CompletableFuture<Boolean> replaceIssueMilestone(TurboIssue issue, Integer newMilestone) {
+        Optional<Integer> originalMilestone = issue.getMilestone();
+
+        logger.info("Changing milestone for " + issue + " on UI");
+        CompletableFuture<Optional<TurboIssue>> localMilestoneReplaceFuture =
+                repoOpControl.replaceIssueMilestoneLocally(issue, newMilestone);
+        localMilestoneReplaceFuture.thenRun(this::refreshUI);
+
+        return updateIssueMilestonesOnServer(issue, newMilestone)
+                .thenCombine(localMilestoneReplaceFuture,
+                        (isUpdateSuccessful, locallyModifiedIssue) -> handleIssueMilestoneUpdateOnServerResult(
+                        isUpdateSuccessful, locallyModifiedIssue, originalMilestone.orElseGet(() -> null)));
```

Comment:
The milestone variables in our TurboIssue & SerializableIssue are all in the Optional form, while the GitHub API uses an integer or null to indicate the presence or absence of a milestone. After this step, all that matters is the the new value being assigned.